npc/re/instances/horror_toy_factory.txt

```diff
@@ -0,0 +1,2025 @@
+		close();
+	}
+
+}
+
+// Original NPC Name: Toy Factory Dimensional Machine
+xmas,233,305,4	script	Dimensional Machine	PORTAL,{
+	if (getstatus(SC_MONSTER_TRANSFORM)) {
+		mes("[Toy Factory Dimensional Machine]");
+		mes("You ^ff0000cannot enter while transformed.^000000");
+		close();
+	}
+	if (BaseLevel < 140) {
+		mes("[Toy Factory Dimensional Machine]");
+		mes("The dimensional machine requires a handful of power to operate. You must be at least Lv 140 to start the machine.");
+		close();
+	}
+	.@party_id = getcharid(CHAR_ID_PARTY);
+	.@p_name$ = getpartyname(.@party_id);
+	.@md_name$ = _("Horror Toy Factory");
+	.@xm_time = questprogress(12331, PLAYTIME);
+	if (.@party_id == 0) {
+		mes("[Toy Factory Dimensional Machine]");
+		mes("Don't you have a companion? You need to form a party to enter.");
+		close();
+	}
+	if (.@xm_time == 0) {
+		if (has_instance("1@xm_d") == "") {
+			mes("[Toy Factory Dimensional Machine]");
+			mes("The dimensional gap is not open yet!");
+		} else if (!.@party_id) {
+			mes("[Toy Factory Dimensional Machine]");
+			mes("Stay where you are.");
+		} else {
+			mapannounce("xmas", sprintf(_$("%s, member of the party %s entered the instance %s."), strcharinfo(PC_NAME), .@p_name$, .@md_name$), bc_map, C_SPRINGGREEN, FW_NORMAL, 12);
+			setquest(12331);
+			warp("1@xm_d", 111, 22);
+			end;
+		}
+	} else if (.@xm_time == 1) {
+		mes("[Toy Factory Dimensional Machine]");
+		mes("Traces of entering the dimensional gap has not disappeared.");
+	} else if (.@xm_time > 1) {
+		mes("^0000ffTraces of entering the dimensional gap has now disappeared. You may now use the Toy Factory Dimensional Machine.^000000");
+		completequest(12331);
+		erasequest(12331);
+	}
+	close();
+}
+
+//-------Area 1--------
+
+1@xm_d,112,20,6	script	Catherine Zeta Jones#0	4_F_SKULL06GIRL,{
+	if (getstatus(SC_MONSTER_TRANSFORM)) {
+		mes("[Catherine Zeta Jones]");
+		mes("You c..cannot enter the instance while transformed.");
+		next();
+		mes("[Catherine Zeta Jones]");
+		mes("Please wait until the transformation w..wears off.");
+		close();
+	}
+	if (getcharid(CHAR_ID_CHAR) == getpartyleader(getcharid(CHAR_ID_PARTY), 2)) {
+		mes("[Catherine Zeta Jones]");
+		mes("There are n..no companions of yours who are under t..transformation right?");
+		npctalk(_("Catherine Zeta Jones: There are n..no companions of yours who are under t..transformation right?"));
+		next();
+		switch (select("End conversation", "Continue talking", "Start now!")) {
+		case 1:
+			mes("[Catherine Zeta Jones]");
+			mes("T..talk to me again when you're ready.");
+			close();
+		case 2:
+			donpcevent(instance_npcname("Catherine Zeta Jones#01")+"::OnStart");
+			close();
+		case 3:
+			donpcevent(instance_npcname("Catherine Zeta Jones#01")+"::OnStart2");
+			close();
+		}
+	} else {
+		mes("[Catherine Zeta Jones]");
+		mes("P..please wait while I am talking to the person in charge.");
+		close();
+	}
+	end;
+}
+
+1@xm_d,112,20,0	script	#bgm01	HIDDEN_WARP_NPC,9,9,{
```

Comment:
is this the official npc name? it look like too common and easy to create conflict in future if there are other npc are used to trigger BGM?
maybe rename with a more unique name ? something like `htf#bgm....`

same goes for other `#bgm` npc...